cv/lane_detection/src/lane_detection_inferencer.cpp

```diff
@@ -0,0 +1,242 @@
+#include "lane_detection/lane_detection_inferencer.h"
+
+#include <algorithm>
+#include <cmath>
+#include <cstdint>
+#include <functional>
+#include <numeric>
+#include <stdexcept>
+#include <vector>
+
+#include "onnxruntime_c_api.h"
+#include "onnxruntime_cxx_api.h"
+#include "opencv2/opencv.hpp"
+
+namespace
+{
+
+constexpr int NETWORK_INPUT_WIDTH{330};
+constexpr int NETWORK_INPUT_HEIGHT{180};
+constexpr int NETWORK_INPUT_CHANNEL{5};
+constexpr int NETWORK_OUTPUT_WIDTH{330};
+constexpr int NETWORK_OUTPUT_HEIGHT{180};
+
+template <typename T> T vectorProduct(const std::vector<T> &v)
+{
+    return std::accumulate(v.begin(), v.end(), 1, std::multiplies<T>());
+}
+
+uint8_t getMedian(std::vector<uint8_t> input)
+{
+    if (input.empty()) {
+        throw std::invalid_argument("Received empty input vector!");
+    }
+
+    std::nth_element(input.begin(), input.begin() + input.size() / 2,
+                     input.end());
+    return input[input.size() / 2];
+}
+
+std::vector<uint8_t> convertToQuadrant(cv::Mat input, const int r_min,
+                                       const int r_max, const int c_min,
+                                       const int c_max)
+{
+    std::vector<uint8_t> output;
+    for (int r = r_min; r < r_max; ++r) {
+        for (int c = c_min; c < c_max; ++c) {
+            output.emplace_back(input.at<uchar>(r, c));
+        }
+    }
+    return output;
+}
+
+cv::Mat getQuadrantEdges(cv::Mat grayscaled_image, const int row_start,
+                         const int row_end, const int col_start,
+                         const int col_end)
+{
+    const std::vector<uint8_t> quad{convertToQuadrant(
+        grayscaled_image, row_start, row_end, col_start, col_end)};
+
+    const double median{static_cast<double>(getMedian(quad))};
+    const double lower_bound{std::max(0.0, std::floor((1 - 0.205) * median))};
+    const double upper_bound{std::min(255.0, std::floor((1 + 0.205) * median))};
+    cv::Mat quad_cv_mat{
+        cv::Mat(quad, CV_8UC1)
+            .reshape(1, static_cast<int>(grayscaled_image.rows / 2))};
+    cv::Mat edges;
+    cv::Canny(quad_cv_mat, edges, lower_bound, upper_bound);
+    return edges;
+}
+
+cv::Mat getEdgeChannel(cv::Mat grayscaled_image)
+{
+    cv::Mat quad1_edges{getQuadrantEdges(
```

Comment:
This part still isn't really clean, but I personally think it's good enough

Thoughts?